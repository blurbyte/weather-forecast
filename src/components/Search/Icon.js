import styled from 'styled-components';

const Wrapper = styled.div`
  position: absolute;
  top: 0.8rem;
  right: 1rem;
  transition: color 0.35s linear;
`;

export default Wrapper;
