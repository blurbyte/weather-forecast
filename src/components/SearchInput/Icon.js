import styled from 'styled-components';

const Icon = styled.div`
  position: absolute;
  z-index: 9;
  top: 0.8rem;
  right: 1rem;
  transition: color 0.35s linear;
`;

export default Icon;
