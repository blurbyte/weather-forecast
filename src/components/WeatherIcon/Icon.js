// Renders appropriate weather conditions icon based on provided weather id

import React from 'react';
import PropTypes from 'prop-types';

import {
  BrokenClouds,
  ClearSkyDay,
  FewClouds,
  Mist,
  Rain,
  ScatteredClouds,
  ShowerRain,
  Snow,
  Thunderstorm
} from '../Icons';

const Icon = ({ weatherId }) => {
  if (weatherId >= 200 && weatherId < 300) {
    return <Thunderstorm />;
  } else if (weatherId >= 300 && weatherId < 400) {
    return <ShowerRain />;
  } else if (weatherId >= 500 && weatherId < 600) {
    return <Rain />;
  } else if (weatherId >= 600 && weatherId < 700) {
    return <Snow />;
  } else if (weatherId >= 700 && weatherId < 800) {
    return <Mist />;
  } else if (weatherId === 801) {
    return <FewClouds />;
  } else if (weatherId === 802) {
    return <ScatteredClouds />;
  } else if (weatherId === 803 || weatherId === 804) {
    return <BrokenClouds />;
  } else {
    return <ClearSkyDay />;
  }
};

Icon.propTypes = {
  weatherId: PropTypes.number.isRequired
};

export default Icon;
