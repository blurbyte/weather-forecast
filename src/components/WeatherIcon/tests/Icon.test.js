import React from 'react';
import renderer from 'react-test-renderer';

import Icon from '../Icon';

jest.mock('../../Icons', () => ({
  BrokenClouds: 'BrokenClouds',
  ClearSkyDay: 'ClearSkyDay',
  FewClouds: 'FewClouds',
  Mist: 'Mist',
  Rain: 'Rain',
  ScatteredClouds: 'ScatteredClouds',
  ShowerRain: 'ShowerRain',
  Snow: 'Snow',
  Thunderstorm: 'Thunderstorm'
}));

test('renders ClearSky', () => {
  const component = renderer.create(<Icon weatherId={800} />);
  expect(component.toJSON()).toMatchSnapshot();
});

test('renders Thunderstorm', () => {
  const component = renderer.create(<Icon weatherId={220} />);
  expect(component.toJSON()).toMatchSnapshot();
});

test('renders ShowerRain', () => {
  const component = renderer.create(<Icon weatherId={314} />);
  expect(component.toJSON()).toMatchSnapshot();
});

test('renders Rain', () => {
  const component = renderer.create(<Icon weatherId={502} />);
  expect(component.toJSON()).toMatchSnapshot();
});

test('renders Snow', () => {
  const component = renderer.create(<Icon weatherId={606} />);
  expect(component.toJSON()).toMatchSnapshot();
});

test('renders Mist', () => {
  const component = renderer.create(<Icon weatherId={714} />);
  expect(component.toJSON()).toMatchSnapshot();
});

test('renders FewClouds', () => {
  const component = renderer.create(<Icon weatherId={801} />);
  expect(component.toJSON()).toMatchSnapshot();
});

test('renders ScatteredClouds', () => {
  const component = renderer.create(<Icon weatherId={802} />);
  expect(component.toJSON()).toMatchSnapshot();
});

test('renders BrokenClouds', () => {
  const component = renderer.create(<Icon weatherId={803} />);
  expect(component.toJSON()).toMatchSnapshot();

  component.update(<Icon weatherId={804} />);
  expect(component.toJSON()).toMatchSnapshot();
});

test('renders ClearSky by default', () => {
  const component = renderer.create(<Icon weatherId={9000} />);
  expect(component.toJSON()).toMatchSnapshot();
});
