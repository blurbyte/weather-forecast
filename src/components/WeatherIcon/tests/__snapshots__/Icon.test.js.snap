// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders BrokenClouds 1`] = `<BrokenClouds />`;

exports[`renders BrokenClouds 2`] = `<BrokenClouds />`;

exports[`renders ClearSky 1`] = `<ClearSkyDay />`;

exports[`renders ClearSky by default 1`] = `<ClearSkyDay />`;

exports[`renders FewClouds 1`] = `<FewClouds />`;

exports[`renders Mist 1`] = `<Mist />`;

exports[`renders Rain 1`] = `<Rain />`;

exports[`renders ScatteredClouds 1`] = `<ScatteredClouds />`;

exports[`renders ShowerRain 1`] = `<ShowerRain />`;

exports[`renders Snow 1`] = `<Snow />`;

exports[`renders Thunderstorm 1`] = `<Thunderstorm />`;
